libr/anal/p/anal_arc.c

```diff
@@ -420,26 +419,28 @@ static int arcompact_op(RAnal *anal, RAnalOp *op, ut64 addr, const ut8 *data, in
 		return 0;
 	}
 
+	op->type = R_ANAL_OP_TYPE_UNK;
+	op->ptr = UT64_MAX;
+	op->val = UT64_MAX;
+	op->jump = UT64_MAX;
+	op->fail = UT64_MAX;
+	op->refptr = 0;
 	op->delay = 0;
 
 	if (anal->big_endian) {
-		int i;
-		for (i = 0; i < 8; i += 4) {
-			words[i / 4] = r_read_be32 (&b[i]);
-		}
+		words[0] = r_read_be32 (&data[0]);
+		words[1] = r_read_be32 (&data[4]);
 	} else {
-		int i;
-		for (i = 0; i < 8; i += 4) {
-			words[i / 4] = r_read_me32 (&b[i]);
-		}
+		words[0] = r_read_me32 (&data[0]);
+		words[1] = r_read_me32 (&data[4]);
 	}
 
-	ut8 opcode = (words[0] & 0xf8000000) >> 27;
+	fields.opcode = (words[0] & 0xf8000000) >> 27;
```

Comment:
The words[] have been endian swapped earlier - using the new standard function for BE, but there is an option to move the so-called "middle-endian" code into the standard functions, too